src/js/game/blueprint.js

```diff
@@ -57,6 +57,37 @@ export class Blueprint {
         return new Blueprint(newEntities);
     }
 
+    /**
+     * Creates a new blueprint from the given entity uids
+     * @param {Array<Entity>} entities
+     */
+    static fromEntities(entities) {
+        const newEntities = [];
+
+        let averagePosition = new Vector();
+
+        // First, create a copy
+        for (let i = entities.length - 1; i >= 0; --i) {
+            const entity = entities[i];
+
+            const clone = entity.clone();
+            newEntities.push(clone);
```

Comment:
Then, assign the value using `i` instead of using `push(clone)` because push spends time handling if multiple arguments were provided to add to the array.

```suggestion
            newEntities[i] = clone;
```

Reply:
Thanks this is a good suggestion!